lumigator/python/mzai/jobs/inference/inference_config.py

```diff
@@ -0,0 +1,39 @@
+from pydantic import BaseModel, ConfigDict
+
+
+class DatasetConfig(BaseModel):
+    path: str
+    model_config = ConfigDict(extra="forbid")
+
+
+class JobConfig(BaseModel):
+    max_samples: int = 0
+    storage_path: str
+    output_field: str = "prediction"
+    enable_tqdm: bool = True
+    model_config = ConfigDict(extra="forbid")
+
+
+class InferenceServerConfig(BaseModel):
+    base_url: str
+    engine: str
+    system_prompt: str | None
+    max_retries: int = 3
+    model_config = ConfigDict(extra="forbid")
+
+
+class SamplingParameters(BaseModel):
+    max_tokens: int = 1024
+    frequency_penalty: float = 0.0
+    temperature: float = 1.0
+    top_p: float = 1.0
+    model_config = ConfigDict(extra="forbid")
+
+
+class InferenceJobConfig(BaseModel):
+    name: str
+    dataset: DatasetConfig
+    job: JobConfig
```

Comment:
Oooh, honest question: what's worse: inheritance or nested configs (even if they are in Pydantic)?

Reply:
Good question!

I think they serve two purposes: e.g. we have a very clear common class (jobs) and I would like all of them to share the base configuration as root parameters, but I'd like other task-specific params (e.g. inference server details for jobs that use it, hf model parameters for those which support them, etc) to be available as subtrees of the main config (mostly to prevent issues with multiple inheritance). 

In both cases the presence of shared classes across jobs, which is something we were not planning to have from the start, might help (intuitively, I think in the case of inheritance not having shared classes might even be counterproductive).